components/kubeflow/kfserving/src/kfservingdeployer.py

```diff
@@ -34,103 +34,232 @@
 from kfserving import V1alpha2InferenceServiceSpec
 from kfserving import V1alpha2InferenceService
 
-def EndpointSpec(framework, storage_uri):
-    if framework == 'tensorflow':
-        return V1alpha2EndpointSpec(predictor=V1alpha2PredictorSpec(tensorflow=V1alpha2TensorflowSpec(storage_uri=storage_uri)))
-    elif framework == 'pytorch':
-        return V1alpha2EndpointSpec(predictor=V1alpha2PredictorSpec(pytorch=V1alpha2PyTorchSpec(storage_uri=storage_uri)))
-    elif framework == 'sklearn':
-        return V1alpha2EndpointSpec(predictor=V1alpha2PredictorSpec(sklearn=V1alpha2SKLearnSpec(storage_uri=storage_uri)))
-    elif framework == 'xgboost':
-        return V1alpha2EndpointSpec(predictor=V1alpha2PredictorSpec(xgboost=V1alpha2XGBoostSpec(storage_uri=storage_uri)))
-    elif framework == 'onnx':
-        return V1alpha2EndpointSpec(predictor=V1alpha2PredictorSpec(onnx=V1alpha2ONNXSpec(storage_uri=storage_uri)))
-    elif framework == 'tensorrt':
-        return V1alpha2EndpointSpec(predictor=V1alpha2PredictorSpec(tensorrt=V1alpha2TensorRTSpec(storage_uri=storage_uri)))
+
+def EndpointSpec(framework, storage_uri, service_account):
```

Comment:
maybe we should set `service_account`'s default to `None` in case we are not passing in the service account argument?

Reply:
We should pass it in. I just missed it again.

Reply:
@Tomcli  the default value will be an empty string, and that will basically mean that no service account will be assigned: 
This 
```json
'spec': {'canary': None,
          'canary_traffic_percent': None,
          'default': {'explainer': None,
                      'predictor': {'custom': None,
                                    'max_replicas': None,
                                    'min_replicas': None,
                                    'onnx': None,
                                    'pytorch': None,
                                    'service_account_name': '',
                                    'sklearn': None,
                                    'tensorflow': {'resources': {'limits': {'cpu': '100m',
                                                                            'memory': '1Gi'},
                                                                 'requests': {'cpu': '100m',
                                                                              'memory': '1Gi'}},
                                                   'runtime_version': None,
                                                   'storage_uri': 'gs://kfserving-samples/models/tensorflow/flowers'},
                                    'tensorrt': None,
                                    'xgboost': None},
                      'transformer': None}} 
```

And since there is no value specified it will not be present upon creation. I verified it.